$NetBSD$

1. On NetBSD it is "MAP_FAILED".
2. mmap returns pointer. Don't compare pointer to integer.

--- src/c/main.d.orig	2009-02-08 06:43:05.000000000 +0300
+++ src/c/main.d	2009-02-08 07:47:59.000000000 +0300
@@ -35,7 +35,11 @@
 #ifdef ECL_USE_MPROTECT
 # include <sys/mman.h>
 # ifndef MAP_FAIL
-#  define MAP_FAIL -1
+#  if defined(MAP_FAILED)
+#   define MAP_FAIL MAP_FAILED
+#  else
+#   define MAP_FAIL ((void *) -1)
+#  endif
 # endif
 #endif
 #include <stdio.h>
